### PR TITLE
Outsource config.ini to appdata

### LINKED ISSUE #36: Hotkey not being saved (back to default after instance restart)
This has something to do with the default directory the program is installed to.

By default, you need admin rights to change files in the directory. So changing the Hotkey in the settings only works for that instance of the program. The changes wont save. 

saving the file in notepad without elevated permissions even leads to this error:
image

A workaround for now is to open the `config.ini` in an editor with elevated permissions and safe the file that way.

Moving the config to `%appdata%` would be the better place tho in the long term, because the program can edit the files there, without needing elevated permissions.

So in layman's terms: the application cant save the changes to the config file.

### PR DESCRIPTION
**Pull request by Chr1s70ph**
This PR reworks how the config is saved and used. Closes #36 

If the QuickKey folder in Appdata is not yet there, the program will create it.
If the `config.ini` File is also not yet in `Appdata/QuickKey`, the program will create it there as well.

Changes to the config will be immediately saved, and ready to use.

### CODE REVIEW

File: src/win/win_path_manager.h [hunk 1]
```diff
@@ -5,12 +5,16 @@
 #include "i_path_manager.h"
 
 class WinPathManager : public IPathManager {
+    private: 
+        const char *appdata = getenv("APPDATA");
+        std::string path = appdata;
```
Comment (leokraft) [inline src/win/win_path_manager.h:10]: I would suggest to rename `path` to `appdata` and directly asigning the result of `getenv`.
Furthermore it would be good to make appdata publicly accessible.

File: src/win/win_path_manager.h [hunk 2]
```diff
@@ -5,12 +5,16 @@
...
+    private: 
+        const char *appdata = getenv("APPDATA");
+        std::string path = appdata;        
     public:
         WinPathManager(){}
         ~WinPathManager(){}
         void initExecutablePath(const std::string called_cmd);
 
         static std::string executablePath;
+        std::string configPath = path.append("\\QuickKey");
```
Comment (leokraft) [inline src/win/win_path_manager.h:17]: Make this static to provide global access. My aim was to prevent creating an instance of the path manager, as to have easy access to all paths.

File: src/win/win_config_manager.h [hunk 3]
```diff
@@ -3,13 +3,22 @@
...
 class WinConfigManager : public IConfigManager {
     public:
-        WinConfigManager(){}
+        WinConfigManager();
         ~WinConfigManager(){}
         void writeConfig(const std::string key, const std::string value);
         std::string getConfig(const std::string key);
+
+    private:
+        void moveConfig();
```
Comment (leokraft) [inline src/win/win_config_manager.h]: What is this function used for? There doesn't seem to be an implementation of it.

File: src/win/win_config_manager.h [hunk 4]
```diff
@@ -3,13 +3,22 @@
...
+        WinPathManager pathManager;
```
Comment (leokraft) [inline src/win/win_config_manager.h]: Change to static access.

File: src/win/win_config_manager.cpp [hunk 5]
```diff
@@ -1,14 +1,46 @@
 #include <windows.h>
 #include "win_config_manager.h"
-#include "win_path_manager.h"
+
+WinConfigManager::WinConfigManager(){
+    
+
+    std::ifstream ifile; //start filereader
+    ifile.open(appdataPath + "\\config.ini"); //open filereader
```
Comment (leokraft) [inline src/win/win_config_manager.cpp:9]: Use `/` instead of `\\`. (if possible)

File: src/win/win_config_manager.h [hunk 6]
```diff
@@ -3,13 +3,22 @@
 
 #include <string>
 #include "i_config_manager.h"
+#include "win_path_manager.h"
```
Comment (leokraft) [inline src/win/win_config_manager.h:5]: Move `includes` needed only in the source file to the source file. Otherwise including this header will always include these libs/files.

**Review by leokraft — CHANGES_REQUESTED** [review]

File: src/win/win_path_manager.h (hunk 1 above)
Comment (leokraft) [inline src/win/win_path_manager.h]: I would suggest to rename `path` to `appdata` and directly asigning the result of `getenv`.
Furthermore it would be good to make appdata publicly accessible.

File: src/win/win_path_manager.h (hunk 2 above)
Comment (leokraft) [inline src/win/win_path_manager.h]: Make this static to provide global access. My aim was to prevent creating an instance of the path manager, as to have easy access to all paths.

**Review by leokraft — CHANGES_REQUESTED** [review]

File: src/win/win_config_manager.h (hunk 3 above)
Comment (Chr1s70ph) [reply src/win/win_config_manager.h]: This is right, there is no implementation. I forgot to remove this out of my commit

File: src/win/win_config_manager.cpp (hunk 5 above)
Comment (Chr1s70ph) [reply src/win/win_config_manager.cpp:9]: appdata is already defined with `\` as seperator for folders. Changing `\\` to `/` would lead to the path being invalid

NOTE: this is due to the function getenv:("APPDATA") in [win_path_manager.h](https://github.com/leokraft/QuickKey/blob/3de7cdccec5a513f77351d9968da055688f1a504/src/win/win_path_manager.h) outputting with `\`

File: src/win/win_path_manager.h (hunk 1 above)
Comment (Chr1s70ph) [reply src/win/win_path_manager.h]: directly assigning is not possible. #getenv() returns a char array, so it has to be initialized as a char. Afterwards it can be converted to a string.

Comment (leokraft) [reply src/win/win_path_manager.h]: `std::string appdata = getenv("APPDATA");` should work just fine
